### ISSUE TITLE
Please use `Model.compile(..., run_eagerly=True)`, or `tf.config.run_functions_eagerly(True)` for more information of where went wrong, or file a issue/bug to `tf.keras`.

### ISSUE THREAD

**Issue by daan0701** [tensorflow/tensorflow#54398]
can't set run_eagerly,but running is wrong:
`model.compile(
        loss=CRFLoss(model.crf, model.dtype),
        optimizer=tf.keras.optimizers.Adam(params["lr"]),
        metrics=[model.crf.viterbi_accuracy, IOBESF1(id2tag)],
        run_eagerly = None)
    model.build((None, train_text.shape[-1]))
    model.summary()

    callbacks = [
        tf.keras.callbacks.ModelCheckpoint(
            filepath=config["ckpt_path"],
            save_weights_only=True,
            save_best_only=True,
            monitor="val_f1",
            mode="max"),
    ]
    model.fit(
        train_dataset,
        epochs=params["epochs"],
        callbacks=callbacks,
        validation_data=dev_dataset)`

**Comment by tilakrayal** [comment]
@daan0701 ,
In order to expedite the trouble-shooting process, could you please provide the complete code, dataset  and the tensorflow version you are using to reproduce the issue reported here.
